File: Java/oicleevan/src/Collatz.java
/*
    collatz.java (collatz java)
    Calculates the collatz conjecture for a given number (in java!!!)
    -- oicleevan
*/

package me.oicleevan.collatz;

import java.util.ArrayList;
import java.util.Scanner;

public class Collatz { 
    
    static ArrayList<Integer> calculate(int num) {
        ArrayList<Integer> terms = new ArrayList<>();
        terms.add(num);

        while (num > 1) {
            if(num % 2 == 1) {
                num = (num * 3) + 1;
            } else {
                num /= 2;
            }
            terms.add(num);
        }

        return terms;
    }
    
    public static void main(String[] args) {
        int n;

        if(args.length > 0) {
            n = Integer.valueOf(args[0]);
        } else {
            Scanner in = new Scanner(System.in);

            System.out.print("Please enter a number: ");
            n = in.nextInt();

            in.close();
        }

        System.out.println("\nRunning the calculator with the number " + n + "...");
        ArrayList<Integer> numbers = calculate(n);

        System.out.println("\nThe numbers are in: " + numbers);
    }
}

File: JavaScript/squivix/collatz.js
function collatz(n) {
    let terms = [n];
    let term = n;
    while (term > 1) {
        if (term % 2)
            term = 3 * term + 1;
        else
            term = Math.floor(term / 2);
        terms.push(term);
    }
    return terms;
}

let n = process.argv[2]
console.log(`Calculating the collatz conjecture from the number ${n}...`)
let list = collatz(n)

console.log("The results are in: ")
console.log(`${list}`)
